Fix some android shared pref warnings

AlexeyBarabash opened the pull request:
<!-- Add brave-browser issue below that this PR will resolve -->
Resolves https://github.com/brave/brave-browser/issues/45453

This PR fixes presubmit warnings from 
https://github.com/brave/brave-core/pull/27986#issuecomment-2803158410 for files 
```
AppearancePreferences.java
BraveHubManagerImpl.java
BraveHubToolbarView.java
```
Except for BraveToolbarManager.java:421, because it uses shared prefs listener.

<!-- CI-related labels that can be applied to this PR:
* CI/run-audit-deps (1) - check for known npm/cargo vulnerabilities (audit_deps)
* CI/run-network-audit (1) - run network-audit
* CI/run-upstream-tests - run Chromium unit and browser tests on Linux and Windows (otherwise only on Linux)
* CI/run-linux-arm64, CI/run-macos-x64, CI/run-windows-arm64, CI/run-windows-x86 - run builds that would otherwise be skipped
* CI/skip - do not run CI builds (except noplatform)
* CI/skip-linux-x64, CI/skip-android, CI/skip-macos-arm64, CI/skip-ios, CI/skip-windows-x64 - skip CI builds for specific platforms
* CI/skip-upstream-tests - do not run Chromium unit, or browser tests (otherwise only on Linux)
* CI/skip-all-linters - do not run presubmit and lint checks
* CI/storybook-url (1) - deploy storybook and provide a unique URL for each build

(1) applied automatically when some files are changed (see: https://github.com/brave/brave-core/blob/master/.github/labeler.yml)
-->

<!--
## Checklist:

- Review design docs
  [Browser design principles](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/chrome_browser_design_principles.md)
  [Style guide](https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++.md)
  [Core principles](https://www.chromium.org/developers/core-principles/)
- Ensure there are (tests)[https://www.chromium.org/developers/testing/]. Unit test as much as possible (including edge cases), but also include browser tests covering high level functionality.
- Ensure that there are comments explaining what classes/methods are/do. The "why" is often more important than the "what" in comments. Also update any relevant docs (moving docs from wiki to brave-core if necessary).
- Request security or other review (third-party libraries, rust code, etc...) if applicable [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) [other review](https://github.com/brave/reviews/issues/new/choose)
  Also see [adding third-party libraries](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/adding_to_third_party.md) for general guidelines on using third party code
- Maks sure there is a [ticket](https://github.com/brave/brave-browser/issues) for your issue
- Use Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- Write a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- Squash any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- Add appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- Run `git rebase master` (if needed)
-->

deeppandya commented on browser/hub/internal/android/java/java_sources.gni:
```diff
@@ -9,3 +9,5 @@ brave_browser_hub_internal_java_sources = [
   "//brave/browser/hub/internal/android/java/src/org/chromium/chrome/browser/hub/BraveHubToolbarView.java",
 ]
+
+brave_browser_hub_internal_java_deps = [ "//chrome/browser/preferences:java" ]
```
Can we create BUILD.gn instead of sources.gni here ?

bridiver replied:
I don't understand what you mean @deeppandya, this is being added as a dep, but I'm not understanding why we need to add an upstream dep to an upstream target?

bridiver replied:
some comments would be helpful here to explain

deeppandya replied:
i meant adding BUILD.gn for the package similar to upsteram instead of sources. but i think it's out of the scope for the PR changes.

AlexeyBarabash replied:
@deeppandya 
the deps I have added is required to provide ability do `import org.chromium.chrome.browser.preferences.ChromeSharedPreferences;`
at https://github.com/brave/brave-core/pull/28679/commits/b59f6dd2a5b01a5f92590562e802b47611f92f9a#diff-ee2bfd2441d8c464a8c54ce22d3a082e13dbf3e9e05e91c9ea08458c280d1588R19

and the `class BraveHubManagerImpl` is involved into bytecode patching at
https://github.com/brave/brave-core/blob/master/build/android/bytecode/java/org/brave/bytecode/BraveHubManagerImplClassAdapter.java#L18

so I don't know, how will it work.

I don't remember, do we have already examples of BraveClass.java and ChromeClass.java which would be placed in different targets, but worked god with bytecode?

bridiver replied:
@deeppandya BUILD.gn for which package? 
@AlexeyBarabash I'm not saying this is wrong necessarily, but it struck me as unusual to add upstream deps to an existing upstream target so some comments would be helpful here

bridiver replied:
Oh, I didn't realize that file existed, this added dep should go in that same file and then it's fine

bridiver replied:
also now I see what @deeppandya means about using a BUILD.gn there instead of sources.gni. It should be its own target and then remove `sources += brave_browser_hub_internal_java_sources`

Then `brave_browser_hub_internal_java_deps = [ "//brave/browser/hub/internal/android/java/src/org/chromium/chrome/browser/hub"]`

bridiver replied:
don't use sources.gni unless there are circular dependencies with the target you are adding it to

AlexeyBarabash replied:
@bridiver 

There are circular dependencies.

I have tried the proposed approach here
https://github.com/brave/brave-core/pull/28679/commits/a701223325910d62ded3db8772b6d9bed11dd7fd

Got circular dependencies error, because:
1) `class BraveHubManagerImpl extends HubManagerImpl` => we must add 
```
deps += [ "//chrome/browser/hub/internal:java" ]
```
into new created `src/brave/browser/hub/internal/BUILD.gn:java`

2) when `BraveHubManagerImpl` sits in its own `BUILD.gn`, we add 
```
deps += brave_browser_hub_internal_java_deps # [ "//brave/browser/hub/internal:java" ]
```
into `src/chrome/browser/hub/internal/BUILD.gn:java`


Tried to use `allow_circular_includes_from` - found it works for *.h/*.cc files.

AlexeyBarabash replied:
here it is how does it look on CI
https://ci.brave.com/job/brave-core-build-pr-android/job/PR-28679/3/console
```
14:54:25  /home/ubuntu/workspace/e-core-build-pr-android_PR-28679/src
14:54:25  > gn gen /home/ubuntu/workspace/e-core-build-pr-android_PR-28679/src/out/android_Component --check
14:54:37  ERROR Dependency cycle:
14:54:37    //brave/browser/hub/internal:java__build_config_crbug_908819 ->
14:54:37    //chrome/browser/hub/internal:java__build_config_crbug_908819 ->
14:54:37    //brave/browser/hub/internal:java__build_config_crbug_908819
```

deeppandya reviewed (approved):
android changes LGTM

AlexeyBarabash commented:
notice from @SergeyZhukovsky at DM
I can get rid of `SharedPreferences.Editor` . For example 
```
preferencesManager.getEditor()
                    .putBoolean()
                    .apply();
```
is the same as `SharedPreferences.writeBoolean`
```
   public void writeBoolean(String key, boolean value) {
        SharedPreferences.Editor ed = getEditor();
        ed.putBoolean(key, value);
        ed.apply();
    }
```

AlexeyBarabash commented:
> Chromium major version is behind target branch (135.0.7049.84 vs 136.0.7103.25). Please rebase.

Rebased.
Removed usage of `SharedPreferences.Editor ` at https://github.com/brave/brave-core/pull/28679/commits/c8b671a20deaaac0701f7954f3be939fd99a1381

bridiver commented on browser/hub/internal/BUILD.gn:
```diff
@@ -0,0 +1,35 @@
+    "//third_party/androidx:androidx_annotation_annotation_java",
+  ]
+
+  # ERROR Dependency cycle:
```
does this comment mean that the code currently doesn't build because of this error?

AlexeyBarabash replied:
@bridiver , yes

bridiver replied:
this is the problem https://github.com/brave/brave-core/pull/28679/files#r2056664974

bridiver replied:
dependencies are compile time, but `HubManagerImpl` doesn't need this at compile time. It just needs to be included somewhere so it's available when asm makes the changes

bridiver commented on browser/hub/internal/android/java/java_sources.gni:
```diff
@@ -3,9 +3,5 @@
-  "//brave/browser/hub/internal/android/java/src/org/chromium/chrome/browser/hub/BraveHubToolbarView.java",
-]
+# Unsuccessfull attempt to get rid brave_browser_hub_internal_java_sources
+brave_browser_hub_internal_java_deps = [ "//brave/browser/hub/internal:java" ]
```
it's not necessary for `chrome/browser/hub/internal` to depend on `brave/browser/hub/internal` because the override is done after the bytecode is generated. `//brave/browser/hub/internal:java` should just be added to `brave_chrome_java_deps`

AlexeyBarabash replied:
thanks lot @bridiver , I didn't see the possibility to use `brave_chrome_java_deps`, addressed the notice at 
https://github.com/brave/brave-core/pull/28679/commits/2796d9bb8afe0220f2637d7b04b7872644627b9c

bridiver commented on patches/chrome-browser-hub-internal-BUILD.gn.patch:
```diff
@@ -1,12 +1,12 @@
-   deps = [
-     ":java_resources",
-     "//base:base_java",
++  deps += brave_browser_hub_internal_java_deps
```
this is not needed as mentioned above

AlexeyBarabash replied:
yeah, I addressed at 
https://github.com/brave/brave-core/pull/28679/commits/2796d9bb8afe0220f2637d7b04b7872644627b9c
but I had to add the patch for visibility at
https://github.com/brave/brave-core/pull/28679/commits/2796d9bb8afe0220f2637d7b04b7872644627b9c#diff-6229f2d1f313458d2032010e20a85231507f66cb9e98ad184bd9af95b3b92cc3R9

AlexeyBarabash commented:
rebased again to resolve conflicts after minor Chromium bump

brave-builds commented:
Released in [v1.80.6](https://github.com/brave/brave-browser/releases/tag/v1.80.6)
